subsys/bluetooth/host/buf.c

```diff
@@ -59,7 +59,7 @@ NET_BUF_POOL_FIXED_DEFINE(evt_pool, CONFIG_BT_BUF_EVT_RX_COUNT,
 			  NULL);
 #else
 NET_BUF_POOL_FIXED_DEFINE(hci_rx_pool, BT_BUF_RX_COUNT,
-			  BT_BUF_RX_SIZE, sizeof(struct bt_buf_data),
+			  BT_BUF_RX_SIZE, sizeof(struct acl_data),
```

Comment:
This is a pretty bad bug. The `hci_acl()` function in `hci_core.c` will write to this struct, which would result in overwriting/currupting the first bytes of the next buffer in the pool (these bytes belong to a linked list pointer). I'm surprised we haven't seen widespread instability caused by this so far. Maybe there's some additional padding somewhere that I missed when analyzing the structs and the code?

Reply:
yep. dunno how it ever worked, we were probably very lucky.